monorepo/monorepo-ws/apps/update-supergraph-app/src/main.ts

```diff
@@ -1,8 +1,37 @@
-import { interval } from "./configuration/config"
-import { SupergraphManager } from "./services"
-import { PrismaManager } from "./services/prisma-manager"
+import {
+  Configuration,
+  GetSupergraphManager,
+  Storage,
+  SupergraphManager,
+} from '@monorepo-ws/supergraph-manager';
+import { interval } from './configuration/config';
+import * as config from './configuration/config';
+import { RoverManager } from './services/rover-manager';
 
-const manager: SupergraphManager = new PrismaManager()
-const runUpdate = setInterval(() => manager.updateSupergraph(), Number(interval))
+const main = async () => {
+  const configuration = getSupergraphConfiguration();
+  const roverManager: RoverManager = createRover(configuration, config.storage);
+  setInterval(() => roverManager.updateSupergraph(), Number(interval));
+};
 
+const getSupergraphConfiguration = (): Configuration => {
+  const supergraphConfiguration: Configuration = {
```

Comment:
tohle přeblejvání je si myslim zbytečný. můžeš předat rovnou celej configuration. on si z toho vyzobe co bude potřebovat. to že tam bude mít nějaký věci navíc nevadí.

hlavně tenhle kód to pak možná v celku zbytečně znepřehledňuje, byť se jedná o triviálnost.

Reply:
předěláno